### ISSUE TITLE
"304 Not Modified" responses should not include Content-Type header

### ISSUE THREAD

**Issue by oppianmatt** [evansd/whitenoise#89]
https://github.com/evansd/whitenoise/issues/85 is related.

When making an If Modified Since request, the 304 response contains a Content-Type header. Not only that but the Content-Type header doesn't match the content which causes issues.

Here a request for a css file get's changed into text/html (which then doesn't render in the browser)

```
$ curl -sD - 'http://localhost:8000/static/pitchup/mobile/css/campsite.css' -H 'accept-encoding: gzip, deflate, sdch' -H 'accept-language: en-GB,en;q=0.8,en-US;q=0.6' -H 'user-agent: Mozilla/5.0 (Linux; Android 5.1.1; Nexus 6 Build/LYZ28E) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.23 Mobile Safari/537.36' -H 'accept: text/css,*/*;q=0.1' --compressed -o /dev/null -H 'If-Modified-Since: Fri, 27 May 2016 11:28:03 GMT'
HTTP/1.1 304 NOT MODIFIED
Server: gunicorn/19.6.0
Date: Fri, 27 May 2016 11:39:57 GMT
Connection: keep-alive
Content-Length: 0
Content-Type: text/html; charset=utf-8
Content-Language: en-gb


```

whitenoise==3.1

**Comment by evansd** [comment]
Ah, thanks for reporting this. This is Django adding a [default Content-Type](https://github.com/django/django/blob/36d36818a30025034cad6f1ee59b2a960a6582ec/django/http/response.py#L56) because I'm using `HttpResponse(status=304)` rather than `HttpNotModifiedResponse`. I wonder if that's also why the Content-Language and Content-Length headers are being added.

Should be easy enough to fix I hope.

**Comment by evansd** [comment]
Fixed in v3.2
